### ISSUE TITLE
Mandate a rebase if upstream failures have since been fixed in master

### ISSUE THREAD

**Issue by kostmo** [kostmo/circleci-failure-tracker#82]
Currently, Dr. CI indicates that failed CI jobs in one's PR that are also failing upstream at the merge base are "not your fault".  Developers currently take this as license to merge.

However, this runs the risk of the upstream failure masking newly-introduced failures in the same CI job.  A possible example is here: https://github.com/pytorch/pytorch/pull/36290#issuecomment-613240682

In this instance, Dr. CI [helpfully suggested](https://github.com/pytorch/pytorch/pull/36290#issuecomment-611273091) that the failures were "not your fault", but the merged PR went on to break the same job in `master`.

<img width="920" alt="Screen Shot 2020-04-13 at 11 02 54 PM" src="https://user-images.githubusercontent.com/261693/79191242-fed5fa00-7dda-11ea-889f-c650964ae672.png">

Sometimes the upstream failure is not fixed by the time the developer needs (due to time pressure) to merge their PR.  However, if the upstream failure has indeed been fixed, we should indicate this to the user and insist that they rebase.

**Comment by ezyang** [comment]
Yes, please do this. This brings us inline with fbcode where we distinguish between "Green" and "No Signal". If there is too much "No Signal" we should ask to rebase.
